dowhy/causal_refuters/add_unobserved_common_cause.py

```diff
@@ -119,634 +123,821 @@ def __init__(self, *args, **kwargs):
+    kappa_t: Optional[Union[float, np.ndarray]] = None,
+    kappa_y: Optional[Union[float, np.ndarray]] = None,
+    frac_strength_treatment: float = 1.0,
+    frac_strength_outcome: float = 1.0,
+    benchmark_common_causes: Optional[List[str]] = None,
+    plot_estimate: bool = True,
+    alpha_s_estimator_list: Optional[List] = None,
+    alpha_s_estimator_param_list: Optional[List[Dict]] = None,
+    g_s_estimator_list: Optional[List] = None,
+    g_s_estimator_param_list: Optional[List[Dict]] = None,
+    plugin_reisz: bool = False,
+) -> Union[PartialLinearSensitivityAnalyzer, NonParametricSensitivityAnalyzer]:
+    """Add an unobserved confounder for refutation using Non-parametric partial R2 methond (Sensitivity Analysis for non-parametric models).
+
+    :param estimate: CausalEstimate: Estimate to run the refutation
+    :param kappa_t: float, numpy.ndarray: Partial R2 of the unobserved confounder wrt the treatment conditioned on the observed confounders. Only in the case of general non-parametric-partial-R2, it is the fraction of variance in the reisz representer that is explained by the unobserved confounder; specifically  (1-r), where r is the ratio of variance of reisz representer, alpha^2, based on observed confounders and that based on all confounders.
+    :param kappa_y: float, numpy.ndarray: Partial R2 of the unobserved confounder wrt the outcome conditioned on the treatment and observed confounders.
+    :param frac_strength_treatment: float: This parameter decides the effect strength of the simulated confounder as a fraction of the effect strength of observed confounders on treatment. Defaults to 1.
+    :param frac_strength_outcome: float: This parameter decides the effect strength of the simulated confounder as a fraction of the effect strength of observed confounders on outcome. Defaults to 1.
+    :param benchmark_common_causes: names of variables for bounding strength of confounders. (relevant only for partial-r2 based simulation methods)
+    :param plot_estimate: Generate contour plot for estimate while performing sensitivity analysis. (default = True).
+                            (relevant only for partial-r2 based simulation methods)
+    :param alpha_s_estimator_list: PENDING_PARAM_DOC
+    :param alpha_s_estimator_param_list: list of dictionaries with parameters for finding alpha_s. (relevant only for non-parametric-partial-R2 simulation method)
+    :param g_s_estimator_list: list of estimator objects for finding g_s. These objects should have fit() and predict() functions implemented. (relevant only for non-parametric-partial-R2 simulation method)
+    :param g_s_estimator_param_list: list of dictionaries with parameters for tuning respective estimators in "g_s_estimator_list". The order of the dictionaries in the list should be consistent with the estimator objects order in "g_s_estimator_list". (relevant only for non-parametric-partial-R2 simulation method)
+    :plugin_reisz: bool: PENDING_PARAM_DOC
```

Comment:
Flag on whether to use the plugin estimator or the nonparametric estimator for reisz representer function (alpha_s).